src/main/java/org/thingsboard/gateway/service/MqttGatewayService.java

```diff
@@ -328,25 +333,12 @@ private void checkDeviceConnected(String deviceName) {
             onDeviceConnect(deviceName);
         }
     }
-
     private MqttDeliveryFuture publishAsync(final String topic, MqttMessage msg, Consumer<IMqttToken> onSuccess, Consumer<Throwable> onFailure) {
-        try {
-            IMqttDeliveryToken token = tbClient.publish(topic, msg, null, new IMqttActionListener() {
-                @Override
-                public void onSuccess(IMqttToken asyncActionToken) {
-                    onSuccess.accept(asyncActionToken);
-                }
-
-                @Override
-                public void onFailure(IMqttToken asyncActionToken, Throwable e) {
-                    onFailure.accept(e);
-                }
-            });
-            return new MqttDeliveryFuture(token);
-        } catch (MqttException e) {
-            onFailure.accept(e);
-            return new MqttDeliveryFuture(e);
+        if (mqttMessageQueue.size() < connection.getMaxQueueSize()) {
+            MqttMessageWrapper messageWrapper = new MqttMessageWrapper(topic, msg, onSuccess, onFailure, System.currentTimeMillis());
+            mqttMessageQueue.add(messageWrapper);
         }
+        return null;
```

Comment:
is this ok?

Reply:
well, I was thinking to change the interface return type to void, as soon as we are decoupling recieving and sending of an mqtt message